File: backend/Models/Session.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models;

[Table("Sessions")]
public class Sessions
{
    [Key]
    public int SessionID { get; set; }

    [Required]
    public int UserID { get; set; }

    [Required]
    [MaxLength(255)]
    public required string SessionType { get; set; }

    public DateTime StartTime { get; set; }

    [Required]
    public int SessionReps { get; set; } = 0;

    public double? SessionScore { get; set; }

    [Column(TypeName = "text")]
    public string? SessionFeedback { get; set; }

    public int SessionDurationSec { get; set; } = 0;

    [ForeignKey("UserID")]
    public User? User { get; set; }

    // Navigation property for reps
    public ICollection<Rep> Reps { get; set; } = new List<Rep>();
}

File: frontend/src/types/session.ts
export interface Session {
  sessionID: number;
  userID: number;
  sessionType: string;
  startTime: string;
  sessionReps: number;
  sessionScore: number | null;
  sessionFeedback: string | null;
  sessionDurationSec: number;
  repScores?: number[];
}

export interface SessionResponse {
  sessionID: number;
  sessionType: string;
  startTime: string;
  sessionReps: number;
  sessionScore: number;
  sessionFeedback: string;
  sessionDurationSec: number;
  repScores: number[];
}

export interface CreateSessionRequest {
  sessionType: string;
  startTime?: string;
  sessionReps?: number;
  sessionScore?: number;
  sessionFeedback?: string;
  sessionDurationSec?: number;
  repScores?: number[];
}

export interface UpdateSessionRequest {
  sessionType?: string;
  startTime?: string;
  sessionReps?: number;
  sessionScore?: number;
  sessionFeedback?: string;
  sessionDurationSec?: number;
  repScores?: number[];
}
